### PR TITLE
remove windows note for dataloader, fix deprecated warning in python3.7

### PR DESCRIPTION
**Pull request by zhreshold**
## Description ##
- multi worker data loader is supported in windows, update the doc correspondingly.
- fix a deprecated feature/bug in python3.7

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by marcoabreu** [comment]
Great! Is this covered by tests or do we maybe have a skip defined somewhere?

**Comment by szha** [comment]
the fixed deprecation is on python 3.7 which is not supported in CI. I think we should start supporting py37 soon.

**Comment by marcoabreu** [comment]
Definitely! My question is whether we test the data loader on windows. Sometimes, we have some conditional skips if we run on windows. It would be great if we could verify that this test is actually run on windows.

**Comment by zhreshold** [comment]
Good point, let me go through the old tests and figure it out

**Comment by zhreshold** [comment]
@marcoabreu This is behavioral test, the test will crash if not supported.

**Comment by marcoabreu** [comment]
Okay, can we then remove the print to reduce the log output and replace it with some no-op? We're already having a lot of useless log (stuff like a function printing the numbers from 1 to 10000) and I'd like to keep that low in future if the log messages don't provide any information about the test itself.

File: tests/python/unittest/test_gluon_data.py [hunk 1]
```diff
@@ -205,24 +206,20 @@ def batchify(self, data):
...
-        data = Dummy(True)
-        loader = DataLoader(data, batch_size=40, batchify_fn=data.batchify_list, num_workers=2)
-        for epoch in range(1):
-            for i, data in enumerate(loader):
-                if i % 100 == 0:
-                    print(data)
-                    print('{}:{}'.format(epoch, i))
+    # sanity tests
+    # enable test on windows
+    data = Dummy(False)
```
Comment (marcoabreu) [inline tests/python/unittest/test_gluon_data.py]: Did you intentionally set the randomized shape to false? It used to be true before.

Also, could elaborate on the comment in line 210? Is there anything specific we have to do on Windows or is it working out of the box? If it doesn't need any special configuration, we can safely remove the comment as it is otherwise confusing to people who don't know the history

Comment (zhreshold) [reply tests/python/unittest/test_gluon_data.py]: - I am deliberately switch one test case to fixed shape, and the rest one is responsible for the random test.
- Will remove the unnecessary comments

BTW, my PR disappeared in CI yesterday but show up today, do you have a clue?

Comment (marcoabreu) [reply tests/python/unittest/test_gluon_data.py]: I see, thanks for elaborating.

Yeah, we had some disk space problems on our server. Everything should be fine now :)

**Comment by zhreshold** [comment]
close due to duplicate efforts in #11908
